### ISSUE TITLE
Add a built-in way to resolve conflicts

### ISSUE THREAD

**Issue by Apostolique** [Apostolique/Apos.Input#1]
Keybind on `Alt` + `Enter` to toggle fullscreen.
In a menu, keybind on `Enter` to select an option.

If a player is currently in a menu, and wants to toggle fullscreen, we need to make sure that the menu option is not accidentally selected.

____

Put a keybind on `Control` + `A` and another keybind on `Control` + `Shift` + `A`. Only one should trigger at a time.

The current way to handle this is to use `AddNot` with a value of `Shift` on the first so that it doesn't trigger. This works when keybinds are hardcoded, but makes it much harder to allow players to choose their own keybinds.

___

In some cases though, it's fine for a single key to trigger more than one keybind. For example, `Shift` + `A` and `Shift` + `B`. Both use `Shift` but both can trigger at the same time on the same frame.

**Comment by Apostolique** [comment]
API proposal:

Add some internal tracking for used input. Example:

```csharp
jump.Pressed();
jump.Pressed();
```

The first one could trigger, the second one could never trigger if the first one did. This would be tracked at the top.

There can be a `SetUsed()` which gets called recursively down the input tree. The state gets cleared during the `UpdateCleanup` step.

The feature could be opt-in from the constructor using a bool (uniqueness):

```csharp
var jump = new KeyboardCondition(Keys.Space, true);
```

**Comment by Apostolique** [comment]
Fixed in cf7bf81
